03week/towersOfHanoi.js

```diff
@@ -13,29 +13,67 @@ let stacks = {
   c: []
 };
 
+const stackA=stacks.a;
+const stackB=stacks.b;
+const stackC=stacks.c;
+const stacksArr=[stackA, stackB, stackC];
+// Create an array that houses all stacks so a forEach function can be applied(shortens code).
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
 function movePiece() {
-  // Your code here
+
+     let startStack.pop();
+     let endStack.push(startStack);
+  //Write a function that allows you to move the blocks(pop/push).
+
+    }
+
+})
+
+});
+
+movePiece();
 
 }
 
 function isLegal() {
-  // Your code here
+
+  forEach (i = 0; i < stacksArr.length-1; i++) {
+       if (stacksArr[i] > stacksArr[i + 1]){
+         const oK=stacksArr.pop();
+       }
+  // Write a test that insures that the piece cannot be stacked on a smaller block.
+  // The value for each piece needs to have a previous piece is less than current piece function.
 
 }
 
 function checkForWin() {
-  // Your code here
+
+    forEach (i = 0; i < stacksArr.length-1; i++) {
```

Comment:
Once again, you shouldn't use loops if absolutely necessary. You could just check if the length of stack b or c has a length of 4